pkg/p2p/peer/router.go

```diff
@@ -54,47 +55,61 @@ func (m *messageRouter) CanRoute(topic topics.Topic) bool {
 	return false
 }
 
-func (m *messageRouter) route(topic topics.Topic, b *bytes.Buffer) {
+func (m *messageRouter) route(b bytes.Buffer, msg message.Message) error {
 	var err error
-	switch topic {
+	category := msg.Category()
+	switch category {
 	case topics.GetBlocks:
-		err = m.blockHashBroker.AdvertiseMissingBlocks(b)
+		topics.Extract(&b)
```

Comment:
Do we need to extract the topic here as well? Since `message.Unmarshal` already extracts this topic.